### ISSUE TITLE
Install loop, empty database, 777 permissions list

### ISSUE THREAD

**Issue by colinmeinke** [ushahidi/Swiftriver-2011#42]
Linux 2.6.9-023stab048.6-enterprise
PHP 5.3.3-1.el5.remi
MYSQL 5.1.48-1.el5.art
Firefox 3.6.8

I'm having trouble installing SwiftRiver properly, I must have tried about 15 times!

I have tried from both 0.2.1 and also the nightly version, with much the same results.

The installer works fine, it passes every step. On completion however it the loops back to the start of the installer. If I manually browse to /web it will work but I am unable to login as the database is empty (no user).

Is there a definative list anywhere of what files need to have 777 permissions, as I'm pretty sure this is the issue, the /web redirect isn't working and the database config files aren't being changed (or at least not in every instance)?

Here is the basic list I always make 777 when installing 0.2.1, on most occassions I also change many more files/folders to 777 (in desperation):
/web/temp.htaccess
/web/application/bootstrap.php
/web/application/cache
/web/application/logs
/core/Configuration/ConfigurationFiles
/core/Modules

Also in 0.2.1 how do you login as there's no link in the top right as the nightly build has?

Thanks,

Colin

**Comment by jgosier** [comment]
The main thing I would suggest is using the nightly, because it's more mature, although it's clearly buggy.  There would be no database, the Swift database is generated on the fly so that's normal.  

0.2.1 has no login because it had no user roles.  That's something that was added later.

I think the problem you're having is that the redirect doesn't work.  Look in the following files:

root/web/application/bootstrap.php

and

.htaccess

Make sure that the redirects point to the url of your instance, It sounds like they aren't getting rewritten properly.  A problem with the installer.

**Comment by colinmeinke** [comment]
I may be wrong, but just retried this. To get it redirecting properly to web all I did was change the index.php file from header("Location: installer/"); to header("Location: web/"); manually.

.htaccess file and bootstrap.php both had base_urls of /root/web as they should, however seeing as they're located within /root/web and and /root/web/application going to /root doesn't ever call them, the only file to read in there is index.php which sends it to /root/installer.

The other issue I get a lot is database access denied error browsing to /root/web/. I guess the database details were't written properly to all files in installation:

I changed and re-uploaded the following files manually:

/root/web/application/modules/riverid/classes/riveridconfig.php
/root/web/application/modules/swiftrivertheming/classes/themingconfig.php
/root/core/Modules/DataContext/MySql_V1/Configuration.xml

Are these the only files that need database details written to?

So changing these gets me as far as a blank swiftriver instance, nothing in the feed, like I'd expect.

If I create a user in the database:

username: 'admin'
password: md5('admin')
role: 'Admin'

I can then log in, but I have no options to add channels or anything actually - only option is log out. This is from the nightly downloaded this morning v0.2.2. This isn't a problem as matt's set me up an instance on the cloud. Just thought it might be of use to you to have some feedback on install.

Thanks,

Colin
